datafusion/src/physical_plan/functions.rs

```diff
@@ -1088,11 +1172,12 @@ fn signature(fun: &BuiltinScalarFunction) -> Signature {
 
     // for now, the list is small, as we do not have many built-in functions.
     match fun {
-        BuiltinScalarFunction::Array => {
-            Signature::Variadic(array_expressions::SUPPORTED_ARRAY_TYPES.to_vec())
-        }
+        BuiltinScalarFunction::Array => Signature::variadic(
+            array_expressions::SUPPORTED_ARRAY_TYPES.to_vec(),
+            Volatility::Immutable,
```

Comment:
better use `fun::volatility()` to get the volatility here so we can manage the volatility in a single place.